wasm.md

```diff
@@ -1452,42 +1465,36 @@ Imports must appear before any allocations in a module, due to validation.
 
 A subtle point is related to tables with inline `elem` definitions: since these may refer to functions by identifier, we need to make sure that tables definitions come after function definitions.
 
-`structureModule` takes a list of definitions and returns a map with different groups of definitions, preserving the order within each group.
+`sortModule` takes a list of definitions and returns a map with different groups of definitions, preserving the order within each group.
 The groups are chosen to represent different stages of allocation and instantiation.
 
 ```k
-    syntax Map ::=  structureModule (     Defns) [function]
-                 | #structureModule (Map, Defns) [function]
-                 | #initialMap ()                [function]
- // -------------------------------------------------------
-    rule #initialMap ()
-      => "typeDecls" |-> .Defns
-         "imports"   |-> .Defns
-         "func/glob" |-> .Defns
-         "allocs"    |-> .Defns
-         "exports"   |-> .Defns
-         "inits"     |-> .Defns
-         "start"     |-> .Defns
+    syntax ModuleDecl ::= sortedModule ( id: OptionalId, types: Defns, importDefns: Defns, funcsGlobals: Defns, allocs: Defns, exports: Defns, inits: Defns, start: Defns )
```

Comment:
Would it help to refine the types of `Defns` further, into FuncDefns, etc? If so I guess we should make sure  any `Defns` list is read-only to get proper subtyping.